### PR TITLE
feat: adding apim for pagopa-selfcare api

### PR DESCRIPTION
**Pull request by andrea-putzu**
### List of changes

Adding APIM for Selfcare pagoPA product Backend

### Type of changes

- [X] Add new resources
- [ ] Update configuration to existing resources
- [ ] Remove existing resources

### Does this introduce a change to production resources with possible user impact?

- [] Yes, users may be impacted applying this change
- [X] No

### Does this introduce an unwanted change on infrastructure? Check terraform plan execution result

- [] Yes
- [X] No

### Other information

---

### CODE REVIEW

**Review by pasqualespica — CHANGES_REQUESTED** [review]

File: src/domains/selc-app/04_apim_selc.tf [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+  api_management_name = local.pagopa_apim_name
+  display_name        = local.apim_selfcare_pagopa_api.display_name
+  versioning_scheme   = "Segment"
+}
+
+
+module "apim_api_selc_api_v1" {
+  source = "git::https://github.com/pagopa/azurerm.git//api_management_api?ref=v2.18.3"
+
+  name                  = format("%s-pagopa-selc-api", local.project)
```
Comment (diegolagospagopa) [inline src/domains/selc-app/04_apim_selc.tf:49]: please use interpolation

File: src/domains/selc-app/04_apim_selc.tf [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    description           = "API to manage institution api key"
+    path                  = "selc/pagopa"
+    subscription_required = false
+    service_url           = null
+  }
+}
+
+resource "azurerm_api_management_api_version_set" "api_selc_api" {
+
+  name                = format("%s-pagopa-selc-api", var.env_short)
```
Comment (diegolagospagopa) [inline src/domains/selc-app/04_apim_selc.tf:38]: please use interpolation

File: src/domains/selc-app/06_selfcare-fe.tf [hunk 3]
```diff
@@ -0,0 +1,300 @@
+/**
+ * selc resource group
+ **/
+resource "azurerm_resource_group" "selc_fe_rg" {
+  count    = var.selc_fe_enabled ? 1 : 0
+  name     = format("%s-selc-fe-rg", local.project)
```
Comment (diegolagospagopa) [inline src/domains/selc-app/06_selfcare-fe.tf:6]: please use interpolation

File: src/domains/selc-app/06_selfcare-fe.tf [hunk 4]
```diff
@@ -0,0 +1,300 @@
...
+  location = var.location
+
+  tags = var.tags
+}
+
+locals {
+  spa = [
+    for i, spa in var.spa :
+    {
+      name  = replace(format("SPA-%s", spa), "-", "")
```
Comment (diegolagospagopa) [inline src/domains/selc-app/06_selfcare-fe.tf:16]: please use interpolation

File: src/domains/selc-app/06_selfcare-fe.tf [hunk 5]
```diff
@@ -0,0 +1,300 @@
...
+      order = i + 3 // +3 required because the order start from 1: 1 is reserved for default application redirect; 2 is reserved for the https rewrite;
+      conditions = [
+        {
+          condition_type   = "url_path_condition"
+          operator         = "BeginsWith"
+          match_values     = [format("/%s/", spa)]
```
Comment (diegolagospagopa) [inline src/domains/selc-app/06_selfcare-fe.tf:22]: please use interpolation

File: src/domains/selc-app/06_selfcare-fe.tf [hunk 6]
```diff
@@ -0,0 +1,300 @@
...
+          condition_type   = "url_file_extension_condition"
+          operator         = "LessThanOrEqual"
+          match_values     = ["0"]
+          negate_condition = false
+          transforms       = null
+        },
+      ]
+      url_rewrite_action = {
+        source_pattern          = format("/%s/", spa)
+        destination             = format("/%s/index.html", spa)
```
Comment (diegolagospagopa) [inline src/domains/selc-app/06_selfcare-fe.tf:36]: please use interpolation

File: src/domains/selc-app/06_selfcare-fe.tf [hunk 7]
```diff
@@ -0,0 +1,300 @@
...
+module "selc_cdn" {
+  source = "git::https://github.com/pagopa/azurerm.git//cdn?ref=v2.12.1"
+  count  = var.selc_fe_enabled ? 1 : 0
+
+  name                = "selc"
+  prefix              = local.project
+  resource_group_name = azurerm_resource_group.selc_fe_rg[0].name
+  location            = var.location
+  #                       selc.<ENV>.platform.pagopa.it
+  hostname              = format("%s.%s.%s", local.dns_zone_selc, local.dns_zone_platform, local.external_domain)
```
Comment (diegolagospagopa) [inline src/domains/selc-app/06_selfcare-fe.tf:60]: please use interpolation

File: src/domains/selc-app/06_selfcare-fe.tf [hunk 8]
```diff
@@ -0,0 +1,300 @@
...
+  https_rewrite_enabled = true
+  lock_enabled          = var.lock_enable
+
+  index_document     = "index.html"
+  error_404_document = "error.html"
+
+  #                             <ENV>.platform.pagopa.it
+  dns_zone_name                = format("%s.%s", local.dns_zone_platform, local.external_domain)
```
Comment (diegolagospagopa) [inline src/domains/selc-app/06_selfcare-fe.tf:68]: please use interpolation

File: src/domains/selc-app/99_variables.tf [hunk 9]
```diff
@@ -21,13 +21,7 @@ variable "env_short" {
...
 
 variable "env" {
-  type        = string
-  default     = ""
-  description = "Contains uat/dev or empty for prod"
-  validation {
-    condition     = contains(["dev", "uat", ""], var.env)
-    error_message = "Allowed values for input_parameter are \"dev\", \"uat\", or \"\" for prod."
-  }
+  type = string
```
Comment (diegolagospagopa) [inline src/domains/selc-app/99_variables.tf:24]: please add description and validation constraint, as in other projects

File: src/domains/selc-app/api/pagopa_token_exchange/jwt_exchange.xml.tpl [hunk 10]
```diff
@@ -16,7 +16,7 @@
         <validate-jwt header-name="IdentityToken" failed-validation-httpcode="401" require-expiration-time="true" require-signed-tokens="true" output-token-variable-name="outputToken">
             <openid-config url="${openid-config-url}" />
             <audiences>
-                <audience>api.pagopa.selfcare.pagopa.it</audience>
+                <audience>api.platform.pagopa.it</audience>
```
Comment (diegolagospagopa) [inline src/domains/selc-app/api/pagopa_token_exchange/jwt_exchange.xml.tpl:19]: can be parametrized?

**Review by diegolagospagopa — CHANGES_REQUESTED** [review]

**Review by pasqualespica — APPROVED** [review]

**Pull request merged**